DashAISoftware/DashAI
# Improve backend code quality

# Summary

This pr improves the quality while avoiding potential bugs of the backend code through the revision of various [ruff](https://github.com/charliermarsh/ruff) linter rules.
Given the latest syntax changes, this pr requires `Python 3.8` to operate properly. 
Also, it requires `ruff>0.0.270` to obtain the same set of used rules.

## Type of change


- Bug fix.
- Refactoring.


## Changes

- Update ruff rules, disabling some unuseful. Some annoying ones were disabled, while new ones were enabled. Check `pyproject.toml` to see ruff enabled rules.
- Change class methods first input from `self` to `cls` (because a class method recieves as a input the class, not the instantiated object).
- Update types definition to python 3.8 specifications.
- Add `from e` to every exception raised (`raise SomException() from e`) inside other exception in a `except` block. This makes the exception displayed by the interpreter can be distinguished between the one you threw versus the one that may have been accidentally thrown by a code error in the except.
- Add several trailing commas in function arguments to re-format the code to improve code legibility (trailing commas cause arguments to be automatically separated into new lines).
- Add context managers to every ill used `open()` in the code.
- Change string contatenations from explicit (i.e., using `+`) to implicit (without `+`, only using newlines).
- Improve several docstrings descriptions.
- Improve several docstring formats with the D ruff rules.
- Update several `pytest.exception` context managers without a match.
- Delete `ABCMeta`from ConfigurableObject (since there is no abstract method to implement).
- Improved some exception messages.
- Delete useless `.keys()` in places where dictionary keys are iterated.
- Merge separated `isinstance` into one. For example `(isinstance(class_column, str) or isinstance(class_column, int))` to `isinstance(class_column, (str, int))`.
- The registers were modified to be compatible with the syntax supported by Python 3.8.

The pr also deletes a `test_sample.py` test file which contained a sample test file not applicable to the project.


## How to Test

No new tests were implemented because no new features were implemented, only bugfixes and refactoring.
Usual: use `pytest .` to check all backend tests.

## Notes

- `flake8` continues to be the default linter for the back and pre-commit, as there are several problems detected by ruff yet to be resolved.
- Depends on #80.
